packages/assets-controllers/src/TokensController.ts

```diff
@@ -254,7 +254,10 @@ export class TokensController extends BaseController<
     });
   }
 
-  _instantiateNewEthersProvider(): any {
+  _instantiateNewEthersProvider(): Web3Provider | undefined {
+    if (this.config.provider === undefined) {
+      return undefined;
+    }
```

Comment:
I know that it *shouldn't* be any other falsey value than `undefined` here but can we still control for it, either by handling it as `undefined` (this makes sense to me) or throw an error?
Also `config` not being an object is handled below so should be here too.
```suggestion
    if (!this.config?.provider) {
      return undefined;
    }
```